src/window/building_info.c

```diff
@@ -565,6 +547,32 @@ static void handle_mouse(const mouse *m)
             }
         }
     }
+    return 0;
+}
+
+static void handle_mouse(const mouse *m)
+{
+    int handled = 0;
+    // general buttons
+    if (context.storage_show_special_orders) {
+        int y_offset = window_building_get_vertical_offset(&context, 28);
+        handled = image_buttons_handle_mouse(m, context.x_offset, y_offset + 400, image_buttons_help_close, 2, &focus_image_button_id);
+    } else {
+        handled = image_buttons_handle_mouse(
+                      m, context.x_offset, context.y_offset + 16 * context.height_blocks - 40,
+                      image_buttons_help_close, 2, &focus_image_button_id);
+    }
+    if (context.can_go_to_advisor) {
+        handled = image_buttons_handle_mouse(
+                      m, context.x_offset, context.y_offset + 16 * context.height_blocks - 40,
+                      image_buttons_advisor, 1, 0);
+    }
+    if (!handled) {
+        handled = handle_specific_building_info_mouse(m);
+    }
+    if (!handled && (m->right.went_up || (m->is_touch && m->left.double_click))) {
```

Comment:
I think second part after && can be moved inside the same if above

Reply:
It can't as `handled` may be changed between both `if`s.

Reply:
You're right, ignore this

Reply:
No problem, thanks for the review ;)